astroid/inference.py

```diff
@@ -562,12 +562,48 @@ def _is_not_implemented(const):
     return isinstance(const, nodes.Const) and const.value is NotImplemented
 
 
+def _infer_old_style_string_formatting(
+    instance: nodes.Const, other: nodes.NodeNG, context: InferenceContext
+) -> tuple[type[util.Uninferable] | nodes.Const]:
+    """Infer the result of '"string" % ...'."""
+    values = None
+    if isinstance(other, nodes.Tuple):
+        inferred_positional = [helpers.safe_infer(i, context) for i in other.elts]
+        if all(isinstance(i, nodes.Const) for i in inferred_positional):
+            values = tuple(i.value for i in inferred_positional)
+    elif isinstance(other, nodes.Dict):
+        values: dict[Any, Any] = {}
+        for pair in other.items:
+            key = helpers.safe_infer(pair[0], context)
+            if not isinstance(key, nodes.Const):
+                return (util.Uninferable,)
```

Comment:
This line is not covered, I guess it's not worth it ?

Reply:
Covered now!